app/controllers/api/v1/allocation_events_controller.rb

```diff
@@ -23,21 +29,41 @@ def create
 
     private
 
+      # TODO: remove constant when allocation `events` endpoint is no longer in use
       PERMITTED_EVENT_PARAMS = [
         :type,
         attributes: %i[timestamp event_name],
       ].freeze
 
+      PERMITTED_CANCEL_PARAMS = [
+        attributes: %i[cancellation_reason cancellation_reason_comment],
+      ].freeze
+
       def validate_params
-        AllocationEvents::ParamsValidator.new(event_params).validate!(action_name.to_sym)
+        # TODO: remove ParamsValidator completely when allocation `events` endpoint is no longer in use
+        AllocationEvents::ParamsValidator.new(event_params).validate!(:cancel)
```

Comment:
we should still be validating parameters, even under the new methods as well

Reply:
The params validator isn't applicable to cancelling as it's checking the format of a timestamp which we don't need and another type parameter which isn't relevant. Instead I'm relying on the model validation rules to ensure that the cancellation reason is set to a valid enum value when the allocation is cancelled - no need for a validator service just for that.

Reply:
Reinstated validation of timestamp for cancel event via the params validator.